### PR TITLE
fix(deo): correct model_native normalization, add classification+flops results

### PR DESCRIPTION
**Pull request by isaaccorley**
Follow-up to #239, which asked for classification/segmentation/flops experiments on the new DEO model in a separate PR.

Found and fixed a real normalization bug before trusting any results: `TorchGeoDEOBench`'s two pathways (`mode="rgb"`, `mode="s2"`) were both using the wrong stats. DEO-FM's own source (`utils/augmentations.py` upstream) ships two different 10-channel Normalize objects — `normalize_hr` (reflectance-scale RGB, for the "HR" pretraining samples where fMoW-Sentinel's optical channels get swapped for co-located aerial RGB from fMoW-RGB) and `normalize_s2` (raw-DN-scale RGB, for genuine single-sensor Sentinel-2). They share the same 7 multispectral values but their RGB channels differ by three orders of magnitude. `mode="rgb"` was using plain ImageNet stats instead of `normalize_hr`'s actual values, and `mode="s2"` was using `normalize_hr`'s RGB plus a per-image max-normalize hack instead of `normalize_s2`'s raw-DN stats — torchgeo's bundled `DEO_Weights.DEO_SWIN.transforms` only carries `normalize_hr`, and that got reused for the s2 pathway without checking it was the right one. Verified against DEO-FM's actual source rather than trusting torchgeo's single bundled transform to cover both cases.

Also fixed the classification-to-dataset routing: `mode="rgb"` isn't a generic "any RGB dataset" pathway, it's DEO's aerial-optical patch embedding specifically (trained on fMoW-RGB). Restricted it to `resisc45` and `m-pv4ger`, the only aerial-only classification datasets in the repo. `mode="s2"` covers the 8 datasets with genuine Sentinel-2 multispectral bands.

Backfills classification (knn5/linear) and flops for both configs, adds regression tests (zero DEO coverage before this), and relocks `uv.lock` for the `torchgeo>=0.10` bump from #239 — that PR bumped the pin but never relocked, so a fresh `uv sync` on main still resolves 0.9.0.

Segmentation intentionally left out: DEO's configs have no `eval.segmentation.layers` yet, holding for a follow-up.

### CODE REVIEW

**Review by nilsleh — DISMISSED** [review]

**Review by nilsleh — APPROVED** [review]

**Comment by blaz-r** [comment]
Cool stuff, thanks for benchmarking DEO. I asked @wolfilip about the normalisation parameters. What's the thing with segmentation? There is a forward-features function in the tgeo model for this case. Do we need to specify some layer names, or how does that work?

**Comment by wolfilip** [comment]
When evaluating the model, I usually use the normalization values provided by the geobench datasets rather than my own pretraining numbers. I use an RGB order instead of the BGR order typical for S2 data. I removed the 60m bands from the S2 data. I don't understand the segmentation config part, but I have a forward_features function in the DEO model that returns the intermediate backbone layers, which I usually use to evaluate segmentation.

**Comment by isaaccorley** [comment]
> When evaluating the model, I usually use the normalization values provided by the geobench datasets rather than my own pretraining numbers. I use an RGB order instead of the BGR order typical for S2 data. I removed the 60m bands from the S2 data. I don't understand the segmentation config part, but I have a forward_features function in the DEO model that returns the intermediate backbone layers, which I usually use to evaluate segmentation.

Thanks for the advice, I'll update and rerun the DEO sweeps!

**Comment by wolfilip** [comment]
I saw the new results, and they are currently unusually low, lower than pretty much any other model I looked at (even just pure Swin), which points to the model not interpreting the evaluation data correctly. When I looked at the code, I see that you use the normalization values for evaluation that I published as used during pretraining. However, when evaluating the model locally on my own, I do something more similar to CROMA: I rescale the weights based on the individual evaluation datasets provided normalization values, so they are roughly in the same distribution as during pretraining (generally [0, 1]). For instance, when doing this for geobench_bigearthnet, I would do something like

`transforms_test = K.AugmentationSequential(
K.Resize(size=(args.input_size, args.input_size)), 
K.Normalize(tuple(avg), tuple(stds)))`

where the normalization avgs and stds are provided by geobench itself from band_stats.json for the specific dataset. I would then delete the 60m bands and reorder the BGR values to RGB (and also their avgs and stds). However, an on-the-fly estimate similar to CROMA would likely work as well.

The published normalization values in https://github.com/torchgeo/torchgeo/blob/main/torchgeo/models/deo.py are used ONLY during pretraining and are there to indicate the data used during pretraining.

I am very sorry for the inconvenience. I should've been clearer initially on how I perform evaluations.

Since the model's focus is mostly on dense multispectral representations, I would appreciate an evaluation on segmentation tasks. If you need any assistance in setting that up, I am willing to offer help.

**Comment by blaz-r** [comment]
@wolfilip tested DEO with the local torchgeo version, and it works much, much better than what we have here. The results in this PR almost seem random.

Does this code, by any chance, use version 0.10.0 from pypi? Because DEO there had a bug and is basically randomly initialised. This was fixed after 0.10.0 went live.

I'm 95% sure that's why the results are like this. @adamjstewart, are there any plans for something like 0.10.1 so this is fixed 😅 

Until then, it's not worth rerunning this at all.

**Comment by adamjstewart** [comment]
We try to do patch releases monthly, so expect 0.10.1 next month.

**Comment by isaaccorley** [comment]
That would make so much sense! Yes we are using torchgeo latest release from pypi not on main. Glad we found the issue!
